src/games/strategy/triplea/ai/proAI/util/ProMatches.java

```diff
@@ -332,6 +332,21 @@ public boolean match(final Territory t) {
     };
   }
 
+  public static Match<Territory> territoryHasNonMobileInfraFactoryAndIsLand() {
+    return new Match<Territory>() {
+      @Override
+      public boolean match(final Territory t) {
+        final Match<Unit> nonMobileinfraFactoryMatch =
+            new CompositeMatchAnd<Unit>(Matches.UnitCanProduceUnits, Matches.UnitIsInfrastructure,
+                Matches.unitHasMovementLeft.invert());
```

Comment:
Gonna hold off on this as invert() is used across the code base significantly. I agree that adding that method would make it more readable but that's potentially doubling the number of Matches. And there are already too many in that one class to ever find anything so might actually make things worse. Matches probably needs a more wholesale refactor. I've tried to at least keep AI only matches in a separate class.